virtcontainers/network.go

```diff
@@ -1277,6 +1278,93 @@ func createVirtualNetworkEndpoint(idx int, ifName string, interworkingModel NetI
 	return endpoint, nil
 }
 
+func generateInterfacesAndRoutes(networkNS NetworkNamespace) ([]*grpc.Interface, []*grpc.Route, error) {
```

Comment:
This code is very Kata agent  specific since it deals with grpc and not applicable for other agents. So I think it should be it in kata agent.

Reply:
I understand, but if you look at the Kata API, we're using both `grpc.Interface` and `grpc.Route` structures as the generic structures to add interfaces or routes.
I think moving this function out of `kata_agent.go` is the first step, and the second logical step would be to move those very generic structure definitions out of `kata-containers/agent/protocols/grpc`. It'd might be better to have them in a separate package so that they don't get associated with the kata-agent.